clippy_lints/src/transmute/missing_transmute_annotations.rs

```diff
@@ -0,0 +1,35 @@
+use rustc_errors::Applicability;
+use rustc_hir::{GenericArg, Path, TyKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty::Ty;
+
+use clippy_utils::diagnostics::span_lint_and_sugg;
+
+use crate::transmute::MISSING_TRANSMUTE_ANNOTATIONS;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, path: &Path<'tcx>, from_ty: Ty<'tcx>, to_ty: Ty<'tcx>) -> bool {
+    let last = path.segments.last().unwrap();
+    let args = last.args;
+    let missing_generic = match args {
+        Some(args) if !args.args.is_empty() => args.args.iter().any(|arg| match arg {
+            GenericArg::Infer(_) => true,
+            GenericArg::Type(ty) => matches!(ty.kind, TyKind::Infer),
+            _ => false,
+        }),
+        _ => true,
+    };
+    if missing_generic {
+        span_lint_and_sugg(
+            cx,
+            MISSING_TRANSMUTE_ANNOTATIONS,
+            last.ident.span.with_hi(path.span.hi()),
+            "transmute used without annotations",
+            "consider adding missing annotations with `transmute::<IN, OUT>`",
```

Comment:
```suggestion
            "consider adding missing annotations",
```
We can shorten it since this is just repeating the suggestion :)

Reply:
:+1: